code/_onclick/click.dm

```diff
@@ -329,12 +329,12 @@ if(selected_ability.target_flags & flagname){\
 		selected_ability.use_ability(A)
 
 /mob/living/carbon/xenomorph/RightClickOn(atom/A)
-	. = ..()
-	if(!selected_ability)
-		return FALSE
-	A = ability_target(A)
-	if(selected_ability.can_use_ability(A))
-		selected_ability.use_ability(A)
+	if(selected_ability) //If we have a selected ability that we can use, return TRUE
+		A = ability_target(A)
+		if(selected_ability.can_use_ability(A))
+			selected_ability.use_ability(A)
+			return TRUE
+	return ..()
```

Comment:
i think you should still call parent in both cases since it calls a signal but not sure

Reply:
above ?

Reply:
Correct this must always parentcall as it has a signal
You could instead of doing this make the selected abilities register a signal on right click that returns one of the flags that cancels the click